util/src/main/java/org/datacommons/util/NameResolver.java

```diff
@@ -0,0 +1,92 @@
+package org.datacommons.util;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.TypedValue;
+import org.datacommons.proto.Mcf.McfGraph.Values;
+import org.datacommons.proto.Mcf.ValueType;
+import org.datacommons.proto.Resolve.ResolveRequest;
+import org.datacommons.proto.Resolve.ResolveResponse;
+import org.datacommons.proto.Resolve.ResolveResponse.Entity.Candidate;
+
+/** Resolves nodes with names by calling the DC resolution API. */
+final class NameResolver {
+  // Names to be resolved.
+  private final Set<String> resolveNames = ConcurrentHashMap.newKeySet();
+
+  // Names that were resolved to DCIDs.
+  // The Map maintains mappings from name to Set of candidate DCIDs.
+  private final ConcurrentHashMap<String, Set<String>> resolvedNames = new ConcurrentHashMap<>();
+
+  private final ReconClient client;
+
+  NameResolver(ReconClient client) {
+    this.client = client;
+  }
+
+  boolean submit(PropertyValues node) {
+    Optional<String> optionalName = getName(node);
+    if (optionalName.isPresent()) {
+      resolveNames.add(optionalName.get());
+      return true;
+    }
+    return false;
+  }
+
+  void drain() {
+    if (!resolveNames.isEmpty()) {
+      ResolveRequest request =
+          ResolveRequest.newBuilder()
+              .addAllNodes(resolveNames)
+              .setProperty("<-description->dcid")
```

Comment:
Resolve based on type would be very useful. https://github.com/datacommonsorg/mixer/blob/master/internal/server/v2/resolve/golden/resolve_description_test.go#L50

Maybe add in follow up.

Reply:
Good point. Added TODO.